### PR TITLE
Send console.log in custom layouts to syslog

### LINKED ISSUE #1294: javascript console log for custom layouts
**Is your feature request related to a problem? Please describe.**
It would be helpful to have a way to see the javascript console log  when developing custom layouts.

**Describe the solution you'd like**
One possibility would be that console.log() content gets written to "syslog", and users can view this with the Mac OS Console app.

**Describe alternatives you've considered**
Another solution could be that the app opens a window/dialog which does nothing other than display the messages that layouts write using "console.log". 

**Additional context**
This is useful for someone developing or troubleshooting a custom layout.

### PR DESCRIPTION
**Pull request by rickythefox**
Hook console.log in custom layouts to write to syslog.

Closes #1294

### CODE REVIEW

File: Amethyst/Layout/Layouts/CustomLayout.swift [hunk 1]
```diff
@@ -49,6 +49,12 @@ class CustomLayout<Window: WindowType>: StatefulLayout<Window> {
             return nil
         }
 
+        context.evaluateScript("var console = { log: function(message) { _consoleLog(message) } }")
+        let consoleLog: @convention(block) (String) -> Void = { message in
+            NSLog(message)
```
Comment (ianyh) [inline Amethyst/Layout/Layouts/CustomLayout.swift]: Can you use the logging used elsewhere rather than directly to syslog?

**Review by ianyh — CHANGES_REQUESTED** [review]

File: Amethyst/Layout/Layouts/CustomLayout.swift (hunk 1 above)
Comment (rickythefox) [reply Amethyst/Layout/Layouts/CustomLayout.swift]:
```    
func applicationDidFinishLaunching(_ notification: Notification) {
#if DEBUG
    log.addDestination(ConsoleDestination())
#endif

log.info("Logging is enabled")
```
This tells me that logging is only enabled for debug builds, right? I don't think using console.log to debug a custom layout should require a debug build of Amethyst?

Comment (ianyh) [reply Amethyst/Layout/Layouts/CustomLayout.swift]: There used to be a command-line option for sending logs to stdout. Looks like I lost it somewhere along the way.

Comment (rickythefox) [reply Amethyst/Layout/Layouts/CustomLayout.swift]: @ianyh added `--log` command line option and switched to the logging framework used by you.

**Review by ianyh — APPROVED** [review]

**Pull request merged**